### PR TITLE
Task lww 56 deposit is using credit card fx payment gateway

### PR DESCRIPTION
**Pull request by asizovitransition**

### CODE REVIEW

File: LykkeApi2/Controllers/ClientController.cs [hunk 1]
```diff
@@ -217,11 +217,18 @@ public class ClientController : Controller
                 return StatusCode((int) HttpStatusCode.InternalServerError);
             }
 
+            var country = (await _personalDataService.GetAsync(_requestContext.ClientId)).Country;
+
+            if (string.IsNullOrEmpty(country))
```
Comment (nikagamkrelidze) [inline LykkeApi2/Controllers/ClientController.cs]: Please move this to Personal Data Service?

File: LykkeApi2/Controllers/FeeSettingsController.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        private readonly IFeeCalculatorClient _feeCalculatorClient;
+
+        public FeeSettingController(IFeeCalculatorClient feeCalculatorClient)
+        {
+            _feeCalculatorClient = feeCalculatorClient ?? throw new ArgumentNullException(nameof(feeCalculatorClient));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Get(string assetId = null)
```
Comment (nikagamkrelidze) [inline LykkeApi2/Controllers/FeeSettingsController.cs]: Do we really need a method for all fees at once? If no, please consider modifying this method, if yes, please consider splitting it in two and returning a model in one case and an array in the other.

File: LykkeApi2/Controllers/BankCardPaymentUrlController.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace LykkeApi2.Controllers
+{
+    [Produces("application/json")]
+    [Authorize]
+    [Route("api/[controller]")]
+    public class BankCardPaymentUrlController : Controller
```
Comment (ozhelenkov) [inline LykkeApi2/Controllers/BankCardPaymentUrlController.cs]: Contract from v1, was not corrrected and discussed

File: LykkeApi2/Controllers/CatalogsController.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace LykkeApi2.Controllers
+{
+    [Authorize]
+    [Produces("application/json")]
+    [Route("api/[controller]")]
+    public class CatalogsController : Controller
```
Comment (ozhelenkov) [inline LykkeApi2/Controllers/CatalogsController.cs]: contract changes again

File: Core/Settings/CreditVouchersSettings.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Core.Settings
```
Comment (rachaelhuber) [inline Core/Settings/CreditVouchersSettings.cs]: Is this class required for the implementation of LWW-56?

File: LykkeApi2/Controllers/CatalogsController.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        private readonly ICountryPhoneCodeService _countryPhoneCodeService;
+
+        public CatalogsController(ICountryPhoneCodeService countryPhoneCodeService)
+        {
+            _countryPhoneCodeService = countryPhoneCodeService;
+        }
+
+        [HttpGet]
+        [SwaggerOperation("GetCountryPhoneCodes")]
```
Comment (galaymv) [inline LykkeApi2/Controllers/CatalogsController.cs]: You need to add an attribute [ProducesResponseType(typeof(CountriesResponseModel), (int)HttpStatusCode.OK)]

File: LykkeApi2/Controllers/ClientController.cs (hunk 1 above)
Comment (rachaelhuber) [reply LykkeApi2/Controllers/ClientController.cs]: We don't need this for LWW-56 - to be removed

**Pull request closed without merging**